clang/lib/3C/ProgramInfo.cpp

```diff
@@ -1034,15 +1034,21 @@ ProgramInfo::getTypeParamBindings(CallExpr *CE, ASTContext *C) const {
   return TypeParamBindings.at(PSL);
 }
 
-std::pair<CVarSet, bool> ProgramInfo::lookupTypedef(PersistentSourceLoc PSL) {
+CVarOption ProgramInfo::lookupTypedef(PersistentSourceLoc PSL) {
   return typedefVars[PSL];
 }
 
 bool ProgramInfo::seenTypedef(PersistentSourceLoc PSL) {
   return typedefVars.count(PSL) != 0;
 }
 
-void ProgramInfo::addTypedef(PersistentSourceLoc PSL, bool ShouldCheck) {
-  CVarSet empty;
-  typedefVars[PSL] = make_pair(empty, ShouldCheck);
+void ProgramInfo::addTypedef(PersistentSourceLoc PSL, bool ShouldCheck,
+                             TypedefDecl* TD, ASTContext &C) {
+  auto Name = "typedef__" + TD->getNameAsString();
+  auto* PV = new PointerVariableConstraint(TD->getUnderlyingType(), nullptr,
+                                       Name, *this, C);
+  if (ShouldCheck && canWrite(PSL.getFileName()))
```

Comment:
While you are improving this condition anyway, can you check for a macro too?  Before or after this PR, 3C crashes for me on the following program:
```c
#define MYDECL typedef int *foo;
MYDECL

void bar(foo p) {}
```